### PR TITLE
Progress bar style fixes + GA events

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

Screenshot 2024-08-15 at 15 22 33

- Style Safe signing modal
- Fix text auto fit for benefits/surplus display on Finished step
- GA tracking events everywhere and on trigger of each step. Including tracking the time from trigger of the `initial` step to order completion.
- Open console log and search for `CowAnalytics` 
- Extra: Adds a tracking event for closing the CowAMM banner

 
Screenshot 2024-08-15 at 15 29 07

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @fairlighteth , great, thank you!

Some issues: 
1. When a surplus is >1USD, the finished modal still does not look good 
surplus
2. Is this OK that 'expired' and 'finished' states are triggered twice? 
expired 2 times
3. (not sure) There are no GA events for Discord and 'help' navigation on the Expired step. Should we add them?
image
4. (same like 3): no GA events for link on every step (placing/solving/etc.). Should we add them? 
ev
5. There is no event when close the bar (press on the arrow at the top)
6. When I close the progress bar, and then I open Account modal, I can see a bunch of GA events there. Is this OK?
account
7. Could you please also check some UI issues I reported https://github.com/cowprotocol/cowswap/pull/4033#issuecomment-2291208722? 

Thanks!

**Comment by fairlighteth** [comment]
@elena-zh Addressed all issues that should have been addressed except for https://github.com/cowprotocol/cowswap/pull/4803

**Comment by elena-zh** [comment]
@fairlighteth , thanks! But the build has failed. Could you please retrigger it again?

**Comment by fairlighteth** [comment]
@elena-zh build should work now

**Comment by elena-zh** [comment]
Hey @fairlighteth , thanks for adding GA event and UI updates!
But I've noticed that some new issues with fonts have appeared in the PR: 
1. selected tokens are in bold
tokens in bold
2. fonts in the token list, swap container (in some fields), in the header differ in comparison with the ones on Prod
font different
header
Screenshot_10
3. when opening Account modal I see a lot of analytics requests fired, but there is no active progress bar there
open a bar
4. Some dark mode issues are still present: 
image

Could you please take a look at these issues?

**Comment by fairlighteth** [comment]
@elena-zh 
1-3: Should be solved
4: The icons are not an issue, it's intended like that. I fixed the opacity on dark mode.

**Comment by fairlighteth** [comment]
Oh sorry forgot to mention, I think the plan is to not render this in the activity modal so this should be addressed in another PR.

**Review by elena-zh — APPROVED** [review]
Approved then!

**Comment by fairlighteth** [comment]
@elena-zh Also addressed this https://github.com/cowprotocol/cowswap/pull/4803

Screenshot 2024-08-19 at 17 37 30

**Comment by elena-zh** [comment]
@fairlighteth , one nitpick: it would be nice to make this message center-aligned in a mobile view: 
center

File: apps/cowswap-frontend/src/legacy/components/TransactionConfirmationModal/DisplayLink.tsx [hunk 1]
```diff
@@ -9,6 +8,7 @@ import { ExternalLinkCustom } from './styled'
 type DisplayLinkProps = {
   id: string | undefined
   chainId: number
+  onClick?: () => void
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/TransactionConfirmationModal/DisplayLink.tsx:11]: Nitpick: use `Command` type for this.

File: apps/cowswap-frontend/src/modules/wallet/containers/WatchAssetInWallet/index.tsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ export type WatchAssetInWalletProps = {
   shortLabel?: boolean
   className?: string
   fallback?: ReactElement
+  onClick?: () => void
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/wallet/containers/WatchAssetInWallet/index.tsx:21]: Same here: `Command`

**Review by alfetopito — APPROVED** [review]

**Comment by fairlighteth** [comment]
@alfetopito addressed your comments

@elena-zh addressed the Safe tx screen:
Screenshot 2024-08-20 at 12 51 55

**Pull request merged**